src/AutomatedCar/Models/AutomatedCar.cs

```diff
@@ -10,7 +10,11 @@ public class AutomatedCar : Car
         public AutomatedCar(int x, int y, string filename)
             : base(x, y, filename)
         {
+            
+
             this.virtualFunctionBus = new VirtualFunctionBus();
+            this.dummySensor = new DummySensor(this.virtualFunctionBus);
+            this.virtualFunctionBus.RegisterComponent(this.dummySensor);
```

Comment:
You don't need to explicitly register the dummySensor to the bus, this way there would be two sensors registered. The constructor in DummySensor calls the base's constructor, which already deals with registration.